apps/oxofocus/app.js

```diff
@@ -0,0 +1,464 @@
+  
+  return -1;
+}
+
+function first_move_was_a_corner() {
+  if (move_count != 1)
+    return false;
+
+  if (board[1] == "X" || board[3] == "X" || board[7] == "X" || board[9] == "X")
+    return true;
+
+  return false;
+}
+
+function first_move_was_a_side() {
+  if (move_count != 1)
+    return false;
+
+  if (board[2] == "X" || board[4] == "X" || board[6] == "X" || board[8] == "X")
+    return true;
+
+  return false;
+}
+
+function player_adjacent_sides(pl) {
+  if (move_count > 3) return false;
+  if (board[2] == pl && board[4] == pl) return true;
+  if (board[2] == pl && board[6] == pl) return true;
+  if (board[8] == pl && board[2] == pl) return true;
+  if (board[8] == pl && board[6] == pl) return true;
+  
+  return false;
+}
+
+function get_adjacent_corner(pl) {
+  if (board[2] == pl && board[4] == pl) return 1;
+  if (board[2] == pl && board[6] == pl) return 3;
+  if (board[8] == pl && board[2] == pl) return 7;
+  if (board[8] == pl && board[6] == pl) return 9;
+
+  return -1;
+}
+
+function player_has_corner_and_centre(pl) {
+  if (board[1] == pl && board[5] == pl) return true;
+  if (board[3] == pl && board[5] == pl) return true;
+  if (board[7] == pl && board[5] == pl) return true;
+  if (board[9] == pl && board[5] == pl) return true;
+  
+  return false;
+}
+
+function first_move_was_the_centre() {
+  if (move_count == 1 && board[5] == "X")
+    return true;
+  return false;
+}
+
+function get_a_side_move() {
+  return select_random_move_from([2,4,6,8]);
+}
+
+function get_a_corner_move() {
+  return select_random_move_from([1,3,7,9]);
+}
+
+function select_random_move_from(mvs) {
+  var len = mvs.length;
+  var rnd = random(len) - 1;
+  return mvs[rnd];
+}
+
+function random(n) {
+  try {
+    return Math.floor((Math.random() * n) + 1);
+  } catch ( e ) { debug("Error: " + this + e.description); } 
+}
+
+function get_free_cells() {
+  var frees = new Array();
+  
+  for (var i in board) {
+    if (i > 0 && cell_is_free(i))
+      frees.push(i);
+  }
+  return frees;
+}
+
+function get_free_sides() {
+  var frees = new Array();
+  var sides = [2,4,6,8];
+
+  for (var i in sides) {
+    if (cell_is_free(sides[i]))
+      frees.push(sides[i]);
+  }
+  return frees;
+}
+
+function get_free_corner() {
+  var frees = new Array();
+  var sides = [1,3,7,9];
+
+  for (var i in sides) {
+    if (cell_is_free(sides[i]))
+      frees.push(sides[i]);
+  }
+  return frees;
+}
+
+function cell_is_free(i) {
+    if (board[i] == "X" || board[i] == "O") return false;
+    return true;
+}
+
+function valid_move(id) {
+    if (cell_is_free(id) == false) {
+	debug("Invalid move, try another cell");
+	return false;
+    }
+    return true;
+}
+
+function set_cell(n, player) {
+    if (player == "X") {
+	board[n] = "X";
+    } else {
+	board[n] = "O";
+    }
+}
+
+Bangle.on('touch', function(zone,e) {
+  x = Object.values(e)[0];
+  y = Object.values(e)[1];
+
+  if (game_state == GAME_OVER) {
+    new_game();
+    return();
```

Comment:
To fix the build we'd need to remove the `()` from this return

```suggestion
    return;
```